### ISSUE TITLE
Spurious test failures in RelayTest

### ISSUE THREAD

**Issue by JayH5** [praekeltfoundation/relay#70]
Occasionally, we see this failure:
```
  1) test starting the application starts everything (RelayTest)
     test/relay_test.exs:75
     match (=) failed
     The following variables were pinned:
       version_info = "1"
     code:  assert %DiscoveryResponse{version_info: ^version_info, resources: resources} = response
     right: %Envoy.Api.V2.DiscoveryResponse{
              canary: false,
              nonce: "",
              resources: [],
              type_url: "type.googleapis.com/envoy.api.v2.Listener",
              version_info: ""
            }
     stacktrace:
       test/relay_test.exs:67: RelayTest.assert_lds_response/3
       test/relay_test.exs:103: (test)
```

**Comment by JayH5** [comment]
Another one (or possibly the same one since the tests were updated):
```
  1) test starting the application starts everything (RelayTest)
     test/relay_test.exs:225
     Assertion with == failed
     code:  assert resources |> Enum.map(fn any_res -> Cluster.decode(any_res.value()) end) == clusters
     left:  []
     right: [%Envoy.Api.V2.Cluster{alt_stat_name: "", circuit_breakers: nil, cleanup_interval: nil, common_http_protocol_options: nil, common_lb_config: nil, connect_timeout: %Google.Protobuf.Duration{nanos: 0, seconds: 5}, dns_lookup_family: 0, dns_refresh_rate: nil, dns_resolvers: [], eds_cluster_config: %Envoy.Api.V2.Cluster.EdsClusterConfig{eds_config: %Envoy.Api.V2.Core.ConfigSource{config_source_specifier: {:api_config_source, %Envoy.Api.V2.Core.ApiConfigSource{api_type: 2, cluster_names: [], grpc_services: [%Envoy.Api.V2.Core.GrpcService{credentials: [], initial_metadata: [], target_specifier: {:google_grpc, %Envoy.Api.V2.Core.GrpcService.GoogleGrpc{config: nil, ssl_credentials: nil, stat_prefix: "xds_cluster", target_uri: "127.0.0.1:5000"}}, timeout: nil}], refresh_delay: nil}}}, service_name: "/mc2_0"}, health_checks: [], hosts: [], http2_protocol_options: nil, http_protocol_options: nil, lb_config: nil, lb_policy: 0, lb_subset_config: nil, max_requests_per_connection: nil, metadata: nil, name: "/mc2_0", outlier_detection: nil, per_connection_buffer_limit_bytes: nil, protocol_selection: 0, tls_context: nil, transport_socket: nil, type: 3, upstream_bind_config: nil}]
     stacktrace:
       test/relay_test.exs:210: RelayTest.assert_cds_response/2
       test/relay_test.exs:253: (test)
```

**Comment by jerith** [comment]
Fixed (hopefully) in #88.
